src/main/java/com/yugabyte/sample/apps/CassandraEventData.java

```diff
@@ -0,0 +1,307 @@
+		long endTs = dataSource.getEndTs();
+
+		// Bind the select statement.
+		BoundStatement select = getPreparedSelect().bind().setString("deviceId", dataSource.getDeviceId())
+				.setLong("startTs", startTs).setLong("endTs", endTs).setString("event_type", dataSource.getEvent_type())
+				.setInt("readBatchSize", appConfig.cassandraReadBatchSize);
+		// Make the query.
+		ResultSet rs = getCassandraClient().execute(select);
+		return 1;
+	}
+
+	private ByteBuffer getValue(String device_id) {
+
+		byte[] randBytesArr = new byte[appConfig.valueSize];
+
+		getRandomValue(device_id.getBytes(), appConfig.valueSize, randBytesArr);
+
+		return ByteBuffer.wrap(randBytesArr);
+	}
+
+	@Override
+	public long doWrite(int threadIdx) {
+		// Pick a random data source.
+		DataSource dataSource = dataSources.get(random.nextInt(dataSources.size()));
+		long numKeysWritten = 0;
+
+		BatchStatement batch = new BatchStatement();
+		// Enter a batch of data points.
+		long ts = dataSource.getDataEmitTs();
+		for (int i = 0; i < appConfig.cassandraBatchSize; i++) {
+			batch.add(getPreparedInsert().bind().setString("device_id", dataSource.getDeviceId()).setLong("ts", ts)
+					.setString("event_type", dataSource.getEvent_type())
+					.setBytesUnsafe("value", getValue(dataSource.getDeviceId())));
+			numKeysWritten++;
+			ts++;
+		}
+		dataSource.setLastEmittedTs(ts);
+
+		getCassandraClient().execute(batch);
+
+		return numKeysWritten;
+	}
+
+	/**
+	 * This class represents a single device data source, which sends back
+	 * timeseries data for that device. It generates data governed by the emit rate.
+	 */
+	public static class DataSource {
+		// The list of devices to emit for this data source.
+		private String device_id;
+		// The timestamp at which the data emit started.
+		private long dataEmitStartTs = 1;
+		// State variable tracking the last time emitted by this source.
+		private long lastEmittedTs = -1;
+		// Event Type of the data record
+		private String event_type;
+
+		public DataSource(int index) {
+			this.device_id = String.format("device-%05d", index);
+		}
+
+		public String getDeviceId() {
+			return device_id;
+		}
+
+		public boolean getHasEmittedData() {
+			return lastEmittedTs >= dataEmitStartTs;
+		}
+
+		public long getEndTs() {
+			return getLastEmittedTs() + 1;
+		}
+
+		public long getStartTs() {
+			return getEndTs() > appConfig.readBackDeltaTimeFromNow ? getEndTs() - appConfig.readBackDeltaTimeFromNow
+					: dataEmitStartTs;
+		}
+
+		public long getDataEmitTs() {
+			if (lastEmittedTs == -1) {
+				lastEmittedTs = dataEmitStartTs;
+			}
+			return lastEmittedTs;
+		}
+
+		public synchronized long getLastEmittedTs() {
+			return lastEmittedTs;
+		}
+
+		public synchronized void setLastEmittedTs(long ts) {
+			if (lastEmittedTs < ts) {
+				lastEmittedTs = ts;
+			}
+		}
+
+		@Override
+		public String toString() {
+			return getDeviceId();
+		}
+
+		public String getEvent_type() {
+			Random rand = new Random();
```

Comment:
nit: I think it's better to use `Random random` as a field in the class and initialize it in the constructor (rather than creating one each time).

Reply:
Yes agreed, made this change and using an existing class variable